format-delimited/src/main/java/io/cdap/plugin/format/delimited/common/DataTypeDetectorUtils.java

```diff
@@ -99,9 +100,11 @@ public static void detectDataTypeOfRowValues(Map<String, Schema> override,
   public static String[] setColumnNames(String rawLine, boolean skipHeader, String delimiter) {
     String[] columnNames;
     if (skipHeader) {
-      columnNames = rawLine.split(delimiter);
+      // String.split uses regex. Here we safely escape regex sequences by using Pattern.quote
+      // Pattern.quote returns a literal pattern string
+      columnNames = rawLine.split(Pattern.quote(delimiter));
```

Comment:
nitpick: If the delimiter will always be used as `Pattern.quote(delimiter)`, consider storing this in a separate variable.

```
String quotedDelimiter = Pattern.quote(delimiter):
...
rawLine.split(quotedDelimiter);
```

Reply:
Yup makes sense.